### ISSUE TITLE
Duplicate identifier 'URL'.

### ISSUE THREAD

**Issue by onlydave** [wmonk/create-react-app-typescript#314]
```
Failed to compile.

/apps/my-app/node_modules/@types/node/index.d.ts
(2381,15): Duplicate identifier 'URL'.
```

### Environment

```
react-scripts-ts@2.15.1
npm -v 6.0.0 (tried @4 too)
node -v v8.11.1
```

create-react-app my-app --scripts-version=react-scripts-ts
cd my-app
npm start

Then got the error above

```
├── @types/jest@22.2.3
├── @types/node@10.0.0
├─┬ @types/react@16.3.12
│ └── csstype@2.4.1
├─┬ @types/react-dom@16.0.5
│ ├── @types/node@10.0.0 deduped
│ └── @types/react@16.3.12 deduped
```

**Comment by onlydave** [comment]
Changing `@types/node` to version `^7.0.21` works. I'll do some trial and error to see if a more recent version works but might not be safe to let create app use the latest version

```
├── @types/jest@22.2.3
├── @types/node@7.0.61
├─┬ @types/react@16.3.12
│ └── csstype@2.4.1
├─┬ @types/react-dom@16.0.5
│ ├── @types/node@7.0.61 deduped
│ └── @types/react@16.3.12 deduped
```

**Comment by ajgassner** [comment]
Same here. Windows 10, node v 10.0.0, npm 4.6.1

**Comment by webstar1208** [comment]
Great Thanks. It works on V9.6.7.
But I have some questions.
I just searched Node previous version but not found 9.6.7.
What 's happened?

**Comment by Guria** [comment]
Node JS 10.0 according to release notes introduces URL as new global

**Comment by Guria** [comment]
So, now it conflicts with typescript lib.d.ts files. And this issue completely unrelated to current repo.
I would recommend to try using typeRoots setting in tsconfig and list only specific node_modules/@types/... needed for you project

**Comment by dereke55** [comment]
Probably a noob question/issue here, but is there a workaround in the meantime?  Attempting to downgrade to `@types/node@9.6.7` as suggested didn't seem to work:

```
$ npx create-react-app test-app  --scripts-version=react-scripts-ts
$ cd test-app
$ npm i @types/node@9.6.7
$ yarn start
```

output:
```
yarn run v1.6.0
$ react-scripts-ts start
/bin/sh: react-scripts-ts: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by goodday2you** [comment]
@dereke55 you have to eject first, then you can manage/change dependencies:
`
npm run eject
npm remove -D @types/node
npm i -D @types/node@9.6.7
`

**Comment by er-aditya** [comment]
Thanks @goodday2you , it works for me, all duplicate errors gone.

**Comment by yajuve** [comment]
I updated version to
`
 npm i @types/node@9.6.7
`
and it works

**Comment by HuangHongRui** [comment]
Thanks..man
It Works for me..
```
npm i @types/node@9.6.7
```

**Comment by onlydave** [comment]
Depends on your version of npm maybe it might not downgrade if you've already got 10.0.0 installed.

The safest thing to do is probably to edit `package.json` and change `@types/node` there. Then `rm -rf node_modules` and then `npm i`

**Comment by dereke55** [comment]
Thanks again @goodday2you , worked like a charm

**Comment by DorianGrey** [comment]
FYI:
I've added a commit to pin the `@types/node` for the moment (though this is not yet released).
https://github.com/wmonk/create-react-app-typescript/commit/10b727fb5a3835b90d4c6e4b74caec173275a53a

However, this might no longer be required regarding this comment:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25356#issuecomment-385538973
The issue with `@types/node` should be fixed with version `10.0.1`. Could you give this version a try (just manually update it) ?

**Comment by wmonk** [comment]
I have just tested a fresh install of `react-scripts-ts` and it works fine installing `@types/node@10.0.1` so am going to close. Please reopen if there is still an issue. Thanks all for helping!
